packages/contracts/deploy/helpers.ts

```diff
@@ -201,45 +239,72 @@ export async function checkPermission(
     data
   );
   if (!isGranted && operation === Operation.Grant) {
-    throw new Error(
-      `(${who.name}: ${who.address}) doesn't have ${permission} on (${where.name}: ${where.address}) in ${permissionManagerContract.address}`
-    );
+    return false;
   }
 
   if (isGranted && operation === Operation.Revoke) {
-    throw new Error(
-      `(${who.name}: ${who.address}) has ${permission} on (${where.name}: ${where.address}) in ${permissionManagerContract.address}`
-    );
+    return false;
   }
+  return true;
 }
 
-export async function managePermission(
+export async function managePermissions(
   permissionManagerContract: ethers.Contract,
   permissions: Permission[]
 ): Promise<void> {
-  const items = permissions.map(permission => [
-    permission.operation,
-    permission.where.address,
-    permission.who.address,
-    permission.condition || ethers.constants.AddressZero,
-    ethers.utils.id(permission.permission),
-  ]);
-
-  const tx = await permissionManagerContract.applyMultiTargetPermissions(items);
-  console.log(`Set permissions with ${tx.hash}. Waiting for confirmation...`);
-  await tx.wait();
+  // filtering permission to only apply those that are needed
+  const items: Permission[] = [];
```

Comment:
Let's do it with `Promise`.  The less transaction better, even if it's read-only tx.